Name: Shrinath Mishra

# **Problem Statement:**
**Building a new Employee Security System**
You, as the Security analyst, at Stark Industries, have been tasked to build a new contactless employee check-in system. Currently the employees use a physical keycard for entry into the building like shown below. 

You have come up with a new idea that uses the employees smartphone and machine learning to provide a contactless system where when an employee enters the firm’s territory, his or her smartphone connects to the server and transmits data from the employee smartphone sensor data like the accelerometer's data. The server performs the calculations and determines this person as
one of the employees using Gait analysis. Essentially it compares the current pattern of the employee’s gait with the historial pattern and if there is a match, the doors automatically open for the employee to walk in.

To test your idea, you have built a dataset of 30 employees and their daily activities here. Design and develop a system that will perform the gait analysis. 

In the given problem statement, we will analyse the 30 employees first on the basis of their daily activities with some pattern. We can name these patterns as the activity labels. There are 6 main patterns in the activity labels as follows:
1. WALKING
2. WALKING_UPSTAIRS
3. WALKING_DOWNSTAIRS
4. SITTING
5. STANDING
6. LAYING

The smartphone with sensors like Accelerometer and Gyroscope will store data of these 6 activity labels. We will then build a machine learning model that will analyse and identify the patterns of an individual and can predict if the person is among the 30 employees that we have registered. 

We will consider different machine learning models that will assist us with their respective accuracies and prediction score, through which we can choose which model has the best accuracy and the prediction score and then proceed ahead to deploy that model into a proper security system.



# **GAIT Analysis**

Gait Analysis is an assessment of the way the body moves from one place to another. In our problem statement through GAIT Analysis, a reliable and precise form of data is stored of an individual while performing their usual activities.

<p align="left">
  <img width="600" height="325" src="https://blogs.mathworks.com/cleve/files/walker.gif">
</p>

In [ ]:
!unzip "/content/UCI HAR Dataset.zip"

Archive:  /content/UCI HAR Dataset.zip
   creating: UCI HAR Dataset/
  inflating: UCI HAR Dataset/.DS_Store  
   creating: __MACOSX/
   creating: __MACOSX/UCI HAR Dataset/
  inflating: __MACOSX/UCI HAR Dataset/._.DS_Store  
  inflating: UCI HAR Dataset/activity_labels.txt  
  inflating: __MACOSX/UCI HAR Dataset/._activity_labels.txt  
  inflating: UCI HAR Dataset/features.txt  
  inflating: __MACOSX/UCI HAR Dataset/._features.txt  
  inflating: UCI HAR Dataset/features_info.txt  
  inflating: __MACOSX/UCI HAR Dataset/._features_info.txt  
  inflating: UCI HAR Dataset/README.txt  
  inflating: __MACOSX/UCI HAR Dataset/._README.txt  
   creating: UCI HAR Dataset/test/
   creating: UCI HAR Dataset/test/Inertial Signals/
  inflating: UCI HAR Dataset/test/Inertial Signals/body_acc_x_test.txt  
   creating: __MACOSX/UCI HAR Dataset/test/
   creating: __MACOSX/UCI HAR Dataset/test/Inertial Signals/
  inflating: __MACOSX/UCI HAR Dataset/test/Inertial Signals/._body_acc_x_test.txt  
  inflating

##**Logistic Regression**

In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np

In [ ]:
train_df = pd.read_csv("UCI HAR Dataset/train/X_train.txt", delim_whitespace = True,names=list(range(0,561)))

train_df['Subject'] = pd.read_csv("UCI HAR Dataset/train/subject_train.txt", header = None, squeeze = True)

train_df["activity"] = pd.read_csv("UCI HAR Dataset/train/y_train.txt", header = None, squeeze = True)

In [ ]:
features = list()
with open("UCI HAR Dataset/features.txt") as f:
    for line in f:
        features.append(line.split()[1])

In [ ]:
features

['tBodyAcc-mean()-X',
 'tBodyAcc-mean()-Y',
 'tBodyAcc-mean()-Z',
 'tBodyAcc-std()-X',
 'tBodyAcc-std()-Y',
 'tBodyAcc-std()-Z',
 'tBodyAcc-mad()-X',
 'tBodyAcc-mad()-Y',
 'tBodyAcc-mad()-Z',
 'tBodyAcc-max()-X',
 'tBodyAcc-max()-Y',
 'tBodyAcc-max()-Z',
 'tBodyAcc-min()-X',
 'tBodyAcc-min()-Y',
 'tBodyAcc-min()-Z',
 'tBodyAcc-sma()',
 'tBodyAcc-energy()-X',
 'tBodyAcc-energy()-Y',
 'tBodyAcc-energy()-Z',
 'tBodyAcc-iqr()-X',
 'tBodyAcc-iqr()-Y',
 'tBodyAcc-iqr()-Z',
 'tBodyAcc-entropy()-X',
 'tBodyAcc-entropy()-Y',
 'tBodyAcc-entropy()-Z',
 'tBodyAcc-arCoeff()-X,1',
 'tBodyAcc-arCoeff()-X,2',
 'tBodyAcc-arCoeff()-X,3',
 'tBodyAcc-arCoeff()-X,4',
 'tBodyAcc-arCoeff()-Y,1',
 'tBodyAcc-arCoeff()-Y,2',
 'tBodyAcc-arCoeff()-Y,3',
 'tBodyAcc-arCoeff()-Y,4',
 'tBodyAcc-arCoeff()-Z,1',
 'tBodyAcc-arCoeff()-Z,2',
 'tBodyAcc-arCoeff()-Z,3',
 'tBodyAcc-arCoeff()-Z,4',
 'tBodyAcc-correlation()-X,Y',
 'tBodyAcc-correlation()-X,Z',
 'tBodyAcc-correlation()-Y,Z',
 'tGravityAcc-mean()-X',
 'tGravityA

In [ ]:
test_df = pd.read_csv("UCI HAR Dataset/test/X_test.txt", delim_whitespace = True,names=list(range(0,561)))

test_df['Subject'] = pd.read_csv("UCI HAR Dataset/test/subject_test.txt", header = None, squeeze = True)

test_df["activity"] = pd.read_csv("UCI HAR Dataset/test/y_test.txt", header = None, squeeze = True)

In [ ]:
train_df.to_csv('train.csv',index_label=False)

In [ ]:
test_df.to_csv('test.csv',index_label=False)

In [ ]:
train_df= pd.read_csv('/content/train.csv')

In [ ]:
train_df.head()

0         1         2         3         4         5         6  \
0  0.288585 -0.020294 -0.132905 -0.995279 -0.983111 -0.913526 -0.995112   
1  0.278419 -0.016411 -0.123520 -0.998245 -0.975300 -0.960322 -0.998807   
2  0.279653 -0.019467 -0.113462 -0.995380 -0.967187 -0.978944 -0.996520   
3  0.279174 -0.026201 -0.123283 -0.996091 -0.983403 -0.990675 -0.997099   
4  0.276629 -0.016570 -0.115362 -0.998139 -0.980817 -0.990482 -0.998321   

          7         8         9  ...       553       554       555       556  \
0 -0.983185 -0.923527 -0.934724  ... -0.710304 -0.112754  0.030400 -0.464761   
1 -0.974914 -0.957686 -0.943068  ... -0.861499  0.053477 -0.007435 -0.732626   
2 -0.963668 -0.977469 -0.938692  ... -0.760104 -0.118559  0.177899  0.100699   
3 -0.982750 -0.989302 -0.938692  ... -0.482845 -0.036788 -0.012892  0.640011   
4 -0.979672 -0.990441 -0.942469  ... -0.699205  0.123320  0.122542  0.693578   

        557       558       559       560  Subject  activity  
0 -0.018446 -0.841247  0.179941 -0.058627        1         5  
1  0.703511 -0.844788  0.180289 -0.054317        1         5  
2  0.808529 -0.848933  0.180637 -0.049118        1         5  
3 -0.485366 -0.848649  0.181935 -0.047663        1         5  
4 -0.615971 -0.847865  0.185151 -0.043892        1         5  

[5 rows x 563 columns]

In [ ]:
test_df= pd.read_csv('/content/test.csv')

In [ ]:
test_df.head()

0         1         2         3         4         5         6  \
0  0.257178 -0.023285 -0.014654 -0.938404 -0.920091 -0.667683 -0.952501   
1  0.286027 -0.013163 -0.119083 -0.975415 -0.967458 -0.944958 -0.986799   
2  0.275485 -0.026050 -0.118152 -0.993819 -0.969926 -0.962748 -0.994403   
3  0.270298 -0.032614 -0.117520 -0.994743 -0.973268 -0.967091 -0.995274   
4  0.274833 -0.027848 -0.129527 -0.993852 -0.967445 -0.978295 -0.994111   

          7         8         9  ...       553       554       555       556  \
0 -0.925249 -0.674302 -0.894088  ... -0.705974  0.006462  0.162920 -0.825886   
1 -0.968401 -0.945823 -0.894088  ... -0.594944 -0.083495  0.017500 -0.434375   
2 -0.970735 -0.963483 -0.939260  ... -0.640736 -0.034956  0.202302  0.064103   
3 -0.974471 -0.968897 -0.938610  ... -0.736124 -0.017067  0.154438  0.340134   
4 -0.965953 -0.977346 -0.938610  ... -0.846595 -0.002223 -0.040046  0.736715   

        557       558       559       560  Subject  activity  
0  0.271151 -0.720009  0.276801 -0.057978        2         5  
1  0.920593 -0.698091  0.281343 -0.083898        2         5  
2  0.145068 -0.702771  0.280083 -0.079346        2         5  
3  0.296407 -0.698954  0.284114 -0.077108        2         5  
4 -0.118545 -0.692245  0.290722 -0.073857        2         5  

[5 rows x 563 columns]

In [ ]:
from sklearn.model_selection import train_test_split
x=train_df.drop('activity',axis=1)
y=train_df['activity']
X_train, X_test, y_train, y_test = train_test_split(x,y,test_size= 0.30)


In [ ]:
X_train

0         1         2         3         4         5         6  \
4149  0.290950 -0.006505 -0.106082 -0.397388  0.018660 -0.313670 -0.441015   
6184  0.353459 -0.040870 -0.090856 -0.315933 -0.174972 -0.285056 -0.366839   
3642  0.278210 -0.019664 -0.115480 -0.995645 -0.982725 -0.992806 -0.995627   
1064  0.228527 -0.012997 -0.106390 -0.963922 -0.942354 -0.966631 -0.967488   
5544  0.277186 -0.017542 -0.105677 -0.997699 -0.991704 -0.987268 -0.997978   
...        ...       ...       ...       ...       ...       ...       ...   
363   0.430209  0.102541 -0.055604 -0.921666 -0.608044 -0.794040 -0.919940   
3476  0.276051 -0.013733 -0.095749 -0.998922 -0.994032 -0.988395 -0.999198   
3959 -0.061041 -0.108200 -0.309738 -0.164307  0.420934  0.090734 -0.225281   
3613  0.280212 -0.009867 -0.107093 -0.998098 -0.975032 -0.976572 -0.998732   
4014  0.281717 -0.014548 -0.102606 -0.994228 -0.992522 -0.982082 -0.995012   

             7         8         9  ...       552       553       554  \
4149 -0.016544 -0.308926 -0.224489  ... -0.208589 -0.559029  0.029955   
6184 -0.169665 -0.255905  0.030036  ... -0.137126 -0.526178 -0.553810   
3642 -0.980386 -0.992834 -0.941328  ... -0.699868 -0.921044 -0.041682   
1064 -0.948001 -0.966005 -0.908369  ... -0.566165 -0.844500 -0.122402   
5544 -0.991203 -0.987673 -0.941686  ... -0.754047 -0.938461 -0.053078   
...        ...       ...       ...  ...       ...       ...       ...   
363  -0.584846 -0.798174 -0.821053  ... -0.248049 -0.607984 -0.056183   
3476 -0.994285 -0.990801 -0.944800  ... -0.743342 -0.927808  0.022586   
3959  0.382560  0.075228 -0.069397  ... -0.392070 -0.735027  0.107721   
3613 -0.974574 -0.977402 -0.939657  ... -0.512339 -0.828512  0.064333   
4014 -0.993172 -0.983224 -0.936999  ... -0.729607 -0.934708 -0.069399   

           555       556       557       558       559       560  Subject  
4149 -0.374770  0.953230  0.795425 -0.439909  0.399985  0.253421       21  
6184  0.612103 -0.312558 -0.652906 -0.686823  0.233410  0.191981       27  
3642 -0.046791  0.369801  0.253295 -0.706053  0.283786 -0.066133       19  
1064 -0.307267  0.003064  0.204270  0.580720 -0.991795  0.006943        6  
5544 -0.773278  0.343479 -0.459415 -0.814723  0.103155 -0.117041       26  
...        ...       ...       ...       ...       ...       ...      ...  
363   0.480826 -0.078209 -0.296734 -0.610380  0.362653 -0.031506        3  
3476 -0.101890  0.208852  0.570727 -0.897975  0.023863  0.015469       17  
3959  0.665769 -0.896951 -0.188358 -0.443914  0.349686  0.306174       19  
3613 -0.029549  0.649491 -0.870583 -0.735001  0.266226 -0.056588       19  
4014 -0.150294  0.433638 -0.205787 -0.733271  0.117606  0.210002       21  

[5146 rows x 562 columns]

In [ ]:
X_test

0         1         2         3         4         5         6  \
399   0.279504 -0.020473 -0.120445 -0.991744 -0.964486 -0.962525 -0.993055   
4030  0.275488 -0.014069 -0.115585 -0.995568 -0.986976 -0.993476 -0.996600   
6599  0.377699 -0.022027 -0.090369  0.232672  0.499351 -0.105920  0.135295   
5034  0.234961 -0.039898 -0.042968  0.124998  0.147127  0.295003 -0.004362   
7225  0.275462 -0.027730 -0.103357 -0.982451 -0.932981 -0.957472 -0.987447   
...        ...       ...       ...       ...       ...       ...       ...   
1634  0.275977 -0.020509 -0.114452 -0.998295 -0.985227 -0.990415 -0.998375   
3225  0.254243 -0.014433 -0.099862 -0.408581 -0.103689 -0.051725 -0.443127   
3324  0.258765 -0.046886 -0.074998 -0.992890 -0.994053 -0.984961 -0.992776   
6834  0.280868 -0.036850 -0.111868 -0.995201 -0.887460 -0.962858 -0.996096   
2480  0.269214 -0.022315 -0.070668 -0.462585 -0.168372 -0.030207 -0.502298   

             7         8         9  ...       552       553       554  \
399  -0.972633 -0.956204 -0.929766  ...  0.057462 -0.424768  0.019418   
4030 -0.987016 -0.993328 -0.933621  ... -0.420867 -0.767997 -0.024524   
6599  0.444180 -0.160987  0.576660  ... -0.395773 -0.775759 -0.709707   
5034  0.132678  0.182438  0.686534  ... -0.447161 -0.739854  0.322552   
7225 -0.937817 -0.958672 -0.899989  ... -0.330392 -0.695157  0.025847   
...        ...       ...       ...  ...       ...       ...       ...   
1634 -0.984414 -0.990143 -0.945243  ... -0.355853 -0.640235  0.090126   
3225 -0.089048 -0.077299 -0.359894  ... -0.308578 -0.670661  0.714955   
3324 -0.993644 -0.984713 -0.946142  ... -0.168504 -0.546304  0.010889   
6834 -0.879117 -0.964532 -0.936444  ... -0.730912 -0.916625  0.058960   
2480 -0.146154 -0.043163 -0.220060  ... -0.537306 -0.811390  0.334880   

           555       556       557       558       559       560  Subject  
399   0.046542  0.079314 -0.117514 -0.877556  0.030503 -0.040474        3  
4030  0.324484  0.307202  0.177998 -0.753780  0.107072  0.196253       21  
6599  0.212475  0.385280  0.056743 -0.709470  0.262463  0.136002       28  
5034 -0.714129  0.950050 -0.431062 -0.774468  0.141859  0.174556       23  
7225 -0.113522  0.253816 -0.288602 -0.741752  0.074984 -0.168224       30  
...        ...       ...       ...       ...       ...       ...      ...  
1634 -0.193626  0.090037 -0.318116 -0.905997  0.142093 -0.032819        8  
3225 -0.551814  0.584951  0.280871 -0.588233 -0.035518  0.278687       16  
3324 -0.455670 -0.263649 -0.042230  0.403940 -0.304912 -0.683110       17  
6834 -0.134417 -0.094606  0.020174 -0.746961  0.010890 -0.142171       29  
2480  0.291757 -0.310405 -0.382085 -0.457895  0.277338  0.352234       14  

[2206 rows x 562 columns]

In [ ]:
X_train.shape, X_test.shape

((5146, 562), (2206, 562))

In [ ]:
y_train.shape, y_test.shape

((5146,), (2206,))

In [ ]:
from sklearn.linear_model import LogisticRegression
model= LogisticRegression(max_iter=5000)
model.fit(X_train, y_train)

LogisticRegression(max_iter=5000)

In [ ]:
model.score(X_train,y_train)

0.9922269724057521

In [ ]:
pred=model.predict(X_test)

In [ ]:
from sklearn.metrics import accuracy_score
accuracy_score(y_test, pred)

0.9845874886672711

In [ ]:
from sklearn.metrics import classification_report
print(classification_report(y_test, pred))

              precision    recall  f1-score   support

           1       1.00      1.00      1.00       364
           2       1.00      1.00      1.00       315
           3       1.00      1.00      1.00       296
           4       0.96      0.95      0.96       377
           5       0.96      0.97      0.96       426
           6       1.00      1.00      1.00       428

    accuracy                           0.98      2206
   macro avg       0.99      0.99      0.99      2206
weighted avg       0.98      0.98      0.98      2206



In [ ]:
from sklearn.metrics import plot_confusion_matrix, confusion_matrix
confusion_matrix(y_test, pred)

array([[363,   1,   0,   0,   0,   0],
       [  0, 315,   0,   0,   0,   0],
       [  0,   0, 296,   0,   0,   0],
       [  0,   0,   0, 358,  19,   0],
       [  0,   0,   0,  14, 412,   0],
       [  0,   0,   0,   0,   0, 428]])

In [ ]:
from sklearn.metrics import f1_score
f1_score(y_test, pred, average='macro')

0.9857457558996786

##**SVC**

In [ ]:
from sklearn import svm

In [ ]:
model = svm.SVC(kernel='rbf')
model.fit(X_train, y_train)

SVC()

In [ ]:
model.score(X_train, y_train)

0.9488923435678197

In [ ]:
pred= model.predict(X_test)

In [ ]:
from sklearn.metrics import accuracy_score
accuracy_score(y_test, pred)

0.942429737080689

In [ ]:
from sklearn.metrics import classification_report
print(classification_report(y_test, pred))

              precision    recall  f1-score   support

           1       0.99      0.99      0.99       364
           2       0.97      0.98      0.98       315
           3       0.99      0.98      0.98       296
           4       0.87      0.83      0.85       377
           5       0.87      0.89      0.88       426
           6       0.99      1.00      0.99       428

    accuracy                           0.94      2206
   macro avg       0.95      0.94      0.95      2206
weighted avg       0.94      0.94      0.94      2206



In [ ]:
from sklearn.metrics import plot_confusion_matrix, confusion_matrix
confusion_matrix(y_test, pred)

array([[360,   3,   1,   0,   0,   0],
       [  4, 309,   2,   0,   0,   0],
       [  1,   6, 289,   0,   0,   0],
       [  0,   0,   0, 314,  57,   6],
       [  0,   0,   0,  47, 379,   0],
       [  0,   0,   0,   0,   0, 428]])

In [ ]:
from sklearn.metrics import f1_score
f1_score(y_test, pred, average='macro')

0.9450481894117501

##**XGBOOST**

In [ ]:
df=pd.read_csv("/content/train.csv")
df.head(10)

0         1         2         3         4         5         6  \
0  0.288585 -0.020294 -0.132905 -0.995279 -0.983111 -0.913526 -0.995112   
1  0.278419 -0.016411 -0.123520 -0.998245 -0.975300 -0.960322 -0.998807   
2  0.279653 -0.019467 -0.113462 -0.995380 -0.967187 -0.978944 -0.996520   
3  0.279174 -0.026201 -0.123283 -0.996091 -0.983403 -0.990675 -0.997099   
4  0.276629 -0.016570 -0.115362 -0.998139 -0.980817 -0.990482 -0.998321   
5  0.277199 -0.010098 -0.105137 -0.997335 -0.990487 -0.995420 -0.997627   
6  0.279454 -0.019641 -0.110022 -0.996921 -0.967186 -0.983118 -0.997003   
7  0.277432 -0.030488 -0.125360 -0.996559 -0.966728 -0.981585 -0.996485   
8  0.277293 -0.021751 -0.120751 -0.997328 -0.961245 -0.983672 -0.997596   
9  0.280586 -0.009960 -0.106065 -0.994803 -0.972758 -0.986244 -0.995405   

          7         8         9  ...       553       554       555       556  \
0 -0.983185 -0.923527 -0.934724  ... -0.710304 -0.112754  0.030400 -0.464761   
1 -0.974914 -0.957686 -0.943068  ... -0.861499  0.053477 -0.007435 -0.732626   
2 -0.963668 -0.977469 -0.938692  ... -0.760104 -0.118559  0.177899  0.100699   
3 -0.982750 -0.989302 -0.938692  ... -0.482845 -0.036788 -0.012892  0.640011   
4 -0.979672 -0.990441 -0.942469  ... -0.699205  0.123320  0.122542  0.693578   
5 -0.990218 -0.995549 -0.942469  ... -0.844619  0.082632 -0.143439  0.275041   
6 -0.966097 -0.983116 -0.940987  ... -0.564430 -0.212754 -0.230622  0.014637   
7 -0.966313 -0.982982 -0.940987  ... -0.421715 -0.020888  0.593996 -0.561871   
8 -0.957236 -0.984379 -0.940598  ... -0.572995  0.012954  0.080936 -0.234313   
9 -0.973663 -0.985642 -0.940028  ...  0.140452 -0.020590 -0.127730 -0.482871   

        557       558       559       560  Subject  activity  
0 -0.018446 -0.841247  0.179941 -0.058627        1         5  
1  0.703511 -0.844788  0.180289 -0.054317        1         5  
2  0.808529 -0.848933  0.180637 -0.049118        1         5  
3 -0.485366 -0.848649  0.181935 -0.047663        1         5  
4 -0.615971 -0.847865  0.185151 -0.043892        1         5  
5 -0.368224 -0.849632  0.184823 -0.042126        1         5  
6 -0.189512 -0.852150  0.182170 -0.043010        1         5  
7  0.467383 -0.851017  0.183779 -0.041976        1         5  
8  0.117797 -0.847971  0.188982 -0.037364        1         5  
9 -0.070670 -0.848294  0.190310 -0.034417        1         5  

[10 rows x 563 columns]

In [ ]:
df.shape

(7352, 563)

In [ ]:
df.columns

Index(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
       ...
       '553', '554', '555', '556', '557', '558', '559', '560', 'Subject',
       'activity'],
      dtype='object', length=563)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7352 entries, 0 to 7351
Columns: 563 entries, 0 to activity
dtypes: float64(561), int64(2)
memory usage: 31.6 MB


In [ ]:
df.isna().sum()

0           0
1           0
2           0
3           0
4           0
           ..
558         0
559         0
560         0
Subject     0
activity    0
Length: 563, dtype: int64

In [ ]:
df=df.drop('Subject',axis=1)
df.head(10)

0         1         2         3         4         5         6  \
0  0.288585 -0.020294 -0.132905 -0.995279 -0.983111 -0.913526 -0.995112   
1  0.278419 -0.016411 -0.123520 -0.998245 -0.975300 -0.960322 -0.998807   
2  0.279653 -0.019467 -0.113462 -0.995380 -0.967187 -0.978944 -0.996520   
3  0.279174 -0.026201 -0.123283 -0.996091 -0.983403 -0.990675 -0.997099   
4  0.276629 -0.016570 -0.115362 -0.998139 -0.980817 -0.990482 -0.998321   
5  0.277199 -0.010098 -0.105137 -0.997335 -0.990487 -0.995420 -0.997627   
6  0.279454 -0.019641 -0.110022 -0.996921 -0.967186 -0.983118 -0.997003   
7  0.277432 -0.030488 -0.125360 -0.996559 -0.966728 -0.981585 -0.996485   
8  0.277293 -0.021751 -0.120751 -0.997328 -0.961245 -0.983672 -0.997596   
9  0.280586 -0.009960 -0.106065 -0.994803 -0.972758 -0.986244 -0.995405   

          7         8         9  ...       552       553       554       555  \
0 -0.983185 -0.923527 -0.934724  ... -0.298676 -0.710304 -0.112754  0.030400   
1 -0.974914 -0.957686 -0.943068  ... -0.595051 -0.861499  0.053477 -0.007435   
2 -0.963668 -0.977469 -0.938692  ... -0.390748 -0.760104 -0.118559  0.177899   
3 -0.982750 -0.989302 -0.938692  ... -0.117290 -0.482845 -0.036788 -0.012892   
4 -0.979672 -0.990441 -0.942469  ... -0.351471 -0.699205  0.123320  0.122542   
5 -0.990218 -0.995549 -0.942469  ... -0.545410 -0.844619  0.082632 -0.143439   
6 -0.966097 -0.983116 -0.940987  ... -0.217198 -0.564430 -0.212754 -0.230622   
7 -0.966313 -0.982982 -0.940987  ... -0.082307 -0.421715 -0.020888  0.593996   
8 -0.957236 -0.984379 -0.940598  ... -0.269401 -0.572995  0.012954  0.080936   
9 -0.973663 -0.985642 -0.940028  ...  0.339526  0.140452 -0.020590 -0.127730   

        556       557       558       559       560  activity  
0 -0.464761 -0.018446 -0.841247  0.179941 -0.058627         5  
1 -0.732626  0.703511 -0.844788  0.180289 -0.054317         5  
2  0.100699  0.808529 -0.848933  0.180637 -0.049118         5  
3  0.640011 -0.485366 -0.848649  0.181935 -0.047663         5  
4  0.693578 -0.615971 -0.847865  0.185151 -0.043892         5  
5  0.275041 -0.368224 -0.849632  0.184823 -0.042126         5  
6  0.014637 -0.189512 -0.852150  0.182170 -0.043010         5  
7 -0.561871  0.467383 -0.851017  0.183779 -0.041976         5  
8 -0.234313  0.117797 -0.847971  0.188982 -0.037364         5  
9 -0.482871 -0.070670 -0.848294  0.190310 -0.034417         5  

[10 rows x 562 columns]

In [ ]:
df['activity'].value_counts()

6    1407
5    1374
4    1286
1    1226
2    1073
3     986
Name: activity, dtype: int64

In [ ]:
from sklearn.model_selection import train_test_split
x=np.array(df.drop('activity',axis=1))
y=np.array(df.activity.tolist())
X_train, X_test, y_train, y_test = train_test_split(x,y,test_size=0.30,random_state=0)

In [ ]:
X_train.shape, y_train.shape

((5146, 561), (5146,))

In [ ]:
X_test.shape, y_test.shape

((2206, 561), (2206,))

In [ ]:
import xgboost as xgb
from xgboost.sklearn import XGBClassifier

In [ ]:
model=XGBClassifier()
model.fit(X_train,y_train)

XGBClassifier(objective='multi:softprob')

In [ ]:
y_pred=model.predict(X_test)

In [ ]:
from sklearn.metrics import accuracy_score
print("Accuracy : "+str(accuracy_score(y_test,y_pred)))

Accuracy : 0.9873073436083409


In [ ]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           1       0.99      0.99      0.99       361
           2       0.99      1.00      1.00       315
           3       1.00      0.99      0.99       280
           4       0.98      0.97      0.97       406
           5       0.97      0.98      0.97       409
           6       1.00      1.00      1.00       435

    accuracy                           0.99      2206
   macro avg       0.99      0.99      0.99      2206
weighted avg       0.99      0.99      0.99      2206



In [ ]:
from sklearn.metrics import confusion_matrix
print("Accuracy : \n"+str(confusion_matrix(y_test,y_pred)))

Accuracy : 
[[359   2   0   0   0   0]
 [  0 315   0   0   0   0]
 [  2   1 277   0   0   0]
 [  0   0   0 392  14   0]
 [  0   0   0   9 400   0]
 [  0   0   0   0   0 435]]


In [ ]:
from sklearn.metrics import f1_score
print("F1-Score : "+str(f1_score(y_test,y_pred,average='macro')))

F1-Score : 0.9879812240424272


| Models Trained | Accuracy Score |
|----------------|----------------|
|Logistic Regression|98.45%                |
|SVC|94.24%               |
|XGBOOST|98.73%               |